packages/react-components/react-swatch-picker-preview/src/components/SwatchPicker/useSwatchPickerState.tsx

```diff
@@ -0,0 +1,34 @@
+import { swatchPickerCSSVars } from './useSwatchPickerStyles.styles';
+import type { SwatchPickerProps, SwatchPickerState } from './SwatchPicker.types';
+
+const { columnCountGrid, cellSize, gridGap } = swatchPickerCSSVars;
+
+export const useSwatchPickerState_unstable = (state: SwatchPickerState, props: SwatchPickerProps) => {
+  const { size = 'medium' } = props;
+
+  const sizeMap = {
+    extraSmall: '20px',
+    small: '24px',
+    medium: '28px',
+    large: '32px',
+  };
+
+  const spacingMap = {
+    small: '2px',
+    medium: '4px',
+  };
+
+  const rootVariables = {
+    [columnCountGrid]: 3,
+    [cellSize]: sizeMap[size],
+    [gridGap]: spacingMap.medium,
```

Comment:
As I see, these are static values, so they could be defined in `makeStyles()` call

Reply:
resolving as these styles will be used from props. But it'll be added in next PR with `layout` prop